include/range/v3/view/exclusive_scan.hpp

```diff
@@ -0,0 +1,164 @@
+/// \file
+// Range v3 library
+//
+//  Copyright Mitsutaka Takeda 2018-present
+//
+//  Use, modification and distribution is subject to the
+//  Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/ericniebler/range-v3
+//
+
+#ifndef RANGE_V3_VIEW_EXCLUSIVE_SCAN_HPP
+#define RANGE_V3_VIEW_EXCLUSIVE_SCAN_HPP
+
+
+#include <range/v3/utility/concepts.hpp>
+#include <range/v3/view_adaptor.hpp>
+#include <range/v3/view/view.hpp>
+
+namespace ranges {
+    inline namespace v3 {
+        /// \addtogroup group-views
+        /// @{
+        template<typename Rng, typename T, typename Fun>
+        struct exclusive_scan_view
+                : view_adaptor<exclusive_scan_view<Rng, T, Fun>, Rng> {
+        private:
+            friend range_access;
+            T init_;
+            semiregular_t<Fun> fun_;
+            using single_pass = SinglePass<iterator_t<Rng>>;
+            using use_sentinel_t = meta::or_<meta::not_<BoundedRange<Rng>>, single_pass>;
+
+            template<bool IsConst>
+            struct adaptor : adaptor_base {
+            private:
+                using exclusive_scan_view_t = meta::const_if_c<IsConst, exclusive_scan_view>;
+                // sum_'s type doesn't have to be range_value_type_t<Rng>.
+                semiregular_t<range_value_type_t<Rng>> sum_;
```

Comment:
Despite that the specification of `exclusive_scan` in the Standard is completely broken, I think the intent of [[exclusive.scan]/3.2](http://eel.is/c++draft/exclusive.scan#3.2):

> All of `binary_op(init, init)`, `binary_op(init, *first)`, and `binary_op(*first, *first)` shall be convertible to `T`.

is that the intermediate values be stored as `T`s, although I could be easily convinced otherwise.